File: Java/SelectionSort.java
/**
 * A class that implements Selection Sort
 */
public class SelectionSort {

    /**
     * Selection Sort Algorithm
     *
     * We keep track of minIndex whose value is less than others.
     * (In the "still unsorted")
     *
     * @param vals: int array
     */
    public static void selectionSort(int[] vals){
        int minIndex;
        for (int i = 0; i < vals.length-1; i++) {
            minIndex = i;
            for (int j = i+1; j < vals.length; j++) {
                if(vals[minIndex] > vals[j]){
                    minIndex = j;
                }
            }
            // helper method to swap values
            swap(vals, minIndex, i);
        }
    }

    private static void swap(int[] vals, int minIndex, int i){
        int temp = vals[i];
        vals[i] = vals[minIndex];
        vals[minIndex] = temp;
    }

    public static void main(String[] args) {
        int[] arr = {7,23,16,50,49,2,71,18};
        System.out.println("Before Selection Sort: ");
        for(int i : arr){
            System.out.print(i+" , ");
        }
        System.out.println("");
        System.out.println("After Selection Sort: ");
        selectionSort(arr);
        for(int i: arr){
            System.out.print(i+" , ");
        }
    }
}

File: Python/Algorithms/selection_sort.py
# Python code for Selection Sort
def selection_sort(L):
    for i in range(len(L)-1):
        min_index = i
        for j in range(i+1, len(L)-1):
            if L[j] < L[min_index]:
                min_index = j
        L[i], L[min_index] = L[min_index], L[i]
        
# Running a test case
L = [3, 1, 41, 59, 26, 53, 59]
print(L)
selection_sort(L)
print(L)        
